bin/pre-commit-hook.js

```diff
@@ -17,7 +18,7 @@ const files = execSync( 'git diff --cached --name-only' )
 	.map( ( name ) => name.trim() )
 	.filter( ( name ) => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
 
-const lintResult = spawnSync( 'eslint-eslines', [ ...files, '--', '--diff=index' ], {
```

Comment:
For the pre-commit hook, we need the `--diff=index` option. Otherwise, it won't work.

Try this testcase:

    git checkout -d testcase-branch
    # break some rule, for ex, editing some spacing in a file
    # make sure your editor doesn't modify what you do (eslint on save, prettier, etc)
    git add modified-file
    git commit # this will return the error and won't commit

It is working for me. Maybe the reason you thought it wasn't working may be that you were doing the commit on the `master` branch instead of using a topic branch. The [behavior defined for calypso](https://github.com/Automattic/wp-calypso/blob/master/.eslines.json) (processor, in eslines terminology) for the `master` branch is different that for the topic branches. In the `master` branch eslines only checks if there are parsing errors (processor `parsing-errors`), but in the topic branches it check for lines modified (processor `lines-modified`).